prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2006,6 +2010,15 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                                 )?
                             );
                         }
+                        "core::f32::<impl f32>::is_nan" |
+                        "core::f64::<impl f64>::is_nan" => {
+                            
+                            // I'd like to replace the function call with this VIR Expr
+                            // let expr = vir::Expr::UnaryOp(UnaryOpKind::IsNaN, args, location);
```

Comment:
The code should be similar to [this](https://github.com/viperproject/prusti-dev/blob/c7cb51791623c12c254f5f8f9c1bc40a58585a8a/prusti-viper/src/encoder/procedure_encoder.rs#L2382-L2391), just instead of `inhaled_expr` you need to use your `UnaryOp`.